hibernate-core/src/main/java/org/hibernate/loader/ast/internal/LoaderSelectBuilder.java

```diff
@@ -419,15 +426,21 @@ private void applyOrdering(TableGroup tableGroup, OrderByFragment orderByFragmen
 		log.tracef( "Starting visitation of FetchParent's Fetchables : %s", fetchParent.getNavigablePath() );
 
 		final List<Fetch> fetches = new ArrayList<>();
+		String fullPath = fetchParent.getNavigablePath().getFullPath();
+		if ( visitedNavigablePath.containsKey( fullPath ) ) {
+			return visitedNavigablePath.get( fullPath );
+		}
```

Comment:
It seems the above code snippet performance could be improved as following:
```
final List<Fetch> fullPathFetches = visitedNavigablePath.get( fullPath );
if ( fullPathFetches != null ) {
    return fullPathFetches;
}
```
It would incur only retrieval once instead of twice. Given that the map value is collection which is never null, we could simplify accordingly. Your code snippet would be mandatory if null is a possible value, though.